pkg/elfparser/elf.go

```diff
@@ -284,10 +284,21 @@ func (e *elfLoader) parseRelocationSection(reloSection *elf.Section, elfFile *el
 func (e *elfLoader) loadProg(loadedProgData map[string]ebpf_progs.CreateEBPFProgInput, loadedMaps map[string]ebpf_maps.BpfMap) (map[string]BpfData, error) {
 
 	loadedprog := make(map[string]BpfData)
+	var pinType uint32
+	for mapName, bpfMap := range loadedMaps {
+		pinType = bpfMap.MapMetaData.PinOptions.Type
```

Comment:
Seems like you are using maps pin option to decide if the program needs pinning? I think we should discuss this..